save_model_output.py

```diff
@@ -3,32 +3,78 @@
-    for dir in OUTPUT_DIRS:
-        for filename in os.listdir(dir):
+    s3 = S3.S3()  # s3 client with methods specific to our needs
+
+    for localDir in OUTPUT_DIRS:
+        bucketSubDirectory: str = flowTemperaturePicker(localDir)
+        for filename in os.listdir(localDir):
             # Send file to backend server if file's timestamp is greater than today
             # Parse timestamp from file
             file_date = datetime.datetime.strptime(filename, "%Y-%m-%d %H.npy")
             # Sets timezone to UTC without affecting other values
             file_date = file_date.replace(tzinfo=datetime.timezone.utc)
+
             if file_date > today:
                 # Read and send file
-                file = open(f"{dir}/{filename}", mode="rb")
-                data = file.read()
-                response = requests.post(BACKEND_URL, files={
-                    "file": data,
-                })
+                fileLocation = f"{localDir}/{filename}"
+                flow = (bucketSubDirectory == "flow")  # if false then file will be uploaded to temperature
+                successful, msg = s3.uploadToS3(fileLocation, filename, flow)
+                if successful:
+                    print(s3.prettyPrint(msg, title="File Upload Response: ")) 
+                else:
+                    print("Upload Failed!")
+    
+    # update contents.json
+    _, response = s3.updateContents()
+    print(response)
+    return None
+
+"""
+For Testing
+
+def save_model_output() -> None:
+    today = datetime.datetime.now(datetime.timezone.utc)
+
+    s3 = S3.S3()  # s3 client with methods specific to our needs
 
-                print(f"Response code: {response}")
+    for localDir in OUTPUT_DIRS:
+        bucketSubDirectory: str = flowTemperaturePicker(localDir)
+        for filename in os.listdir(localDir):
+            # Send file to backend server if file's timestamp is greater than today
+            # Parse timestamp from file
+            file_date = datetime.datetime.strptime(filename, "%Y-%m-%d %H.npy")
+            # Sets timezone to UTC without affecting other values
+            file_date = file_date.replace(tzinfo=datetime.timezone.utc)
+
+            if file_date < today:
+                # Read and send file
+                fileLocation = f"{localDir}/{filename}"
+                flow = (bucketSubDirectory == "flow")  # if false then file will be uploaded to temperature
+                successful, msg = s3.uploadToS3(fileLocation, filename, flow)
+                if successful:
+                    print(s3.prettyPrint(msg, title="File Upload Response: ")) 
+                else:
+                    print("Upload Failed!")
+    
+    # update contents.json
+    _, response = s3.updateContents()
+    print(response)
+    
+    return None
+"""
 
-                # Finished with file
-                file.close()
+def flowTemperaturePicker(directoryPath: str) -> str:
```

Comment:
Give this function a better name; the function has flow and temperature in the name but has generic code inside. Maybe something like `getLastDirectoryInPath`?